src/client/components/pages/entities/cb-review.js

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2022       Ansh Goyal
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import {faPlus, faRotate} from '@fortawesome/free-solid-svg-icons';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import {Rating} from 'react-simple-star-rating';
+import React from 'react';
+import _ from 'lodash';
+import request from 'superagent';
+
+
+const {Button} = bootstrap;
+
+
+const REVIEW_CONTENT_PREVIEW_LENGTH = 75;
+
+function ReviewCard({reviewData}) {
+	const publishedDate = new Date(reviewData.published_on).toDateString();
+	let reviewText = reviewData.text;
+	if (reviewText.length > REVIEW_CONTENT_PREVIEW_LENGTH) {
```

Comment:
This throws an error if there is no review or no text:
`Unhandled rejection TypeError: Cannot read properties of null (reading 'length')`

This should do the trick:
```suggestion
	if (reviewText?.length > REVIEW_CONTENT_PREVIEW_LENGTH) {
```